node/src/neighborhood/gossip_acceptor.rs

```diff
@@ -1367,6 +1470,213 @@ mod tests {
         // ZeroHop is decentralized and should never appear in GossipAcceptor tests
     }
 
+    #[test]
+    fn properly_constructed_standard_ipchange_is_identified_and_handled() {
+        let old_addr = NodeAddr::new (&IpAddr::from_str("1.2.3.4").unwrap(), &vec![2000, 2001]);
+        let new_addr = NodeAddr::new (&IpAddr::from_str("2.3.4.5").unwrap(), &vec![2000, 2001]);
+        let gossip_source = SocketAddr::from (new_addr.clone());
+        let (gossip, new_node, mut db) =
+            make_ipchange (2345, Some(old_addr.clone()), Some(new_addr.clone()),
+                           Mode::Standard);
+        let cryptde = CryptDENull::from(db.root().public_key(), TEST_DEFAULT_CHAIN);
+        let agrs_vec: Vec<AccessibleGossipRecord> = gossip.try_into().unwrap();
+        let (cpm_recipient, _) = make_cpm_recipient();
+        let (stream_handler_pool, _, stream_handler_pool_recording_arc) = make_recorder();
+        let (dispatcher, _, _) = make_recorder();
+        let stream_handler_pool_sub = stream_handler_pool.start().recipient();
+        let dispatcher_sub = dispatcher.start().recipient();
+        let subject = IpChangeHandler::new(stream_handler_pool_sub,
+            dispatcher_sub.clone());
+        let system = System::new("test");
+
+        let qualifies_result =
+            subject.qualifies(&db, &agrs_vec.as_slice(), gossip_source);
+        let handle_result = subject.handle(
+            &cryptde,
+            &mut db,
+            agrs_vec,
+            gossip_source,
+            &vec![],
+            &cpm_recipient,
+        );
+
+        System::current().stop();
+        system.run();
+        assert_eq!(Qualification::Matched, qualifies_result);
+        assert_eq!(
+            handle_result,
+            GossipAcceptanceResult::Absorbed,
+        );
+        let db_node_node_addr = db.node_by_key(new_node.public_key()).unwrap().node_addr_opt().unwrap();
+        assert_eq!(
+            db_node_node_addr,
+            new_addr
+        );
+        let stream_handler_pool_recording = stream_handler_pool_recording_arc.lock().unwrap();
+        assert_eq!(
+            stream_handler_pool_recording.get_record::<RemoveStreamMsg>(0),
+            &RemoveStreamMsg {
+                local_addr: SocketAddr::from (db.root().node_addr_opt().unwrap()),
+                peer_addr: SocketAddr::new(old_addr.ip_addr(), old_addr.ports()[0]),
+                stream_type: RemovedStreamType::Clandestine,
+                sub: dispatcher_sub.clone()
+            }
+        );
+        assert_eq!(
+            stream_handler_pool_recording.get_record::<RemoveStreamMsg>(1),
+            &RemoveStreamMsg {
```

Comment:
If I were you I would write a closure with a single parameter somewhere here containing this whole `RemoveStreamMsg` and which would alter the single parameter in the constructor of SocketAddr for `peer_addr` by its argument passed in. I usually tend to do that for bigger code blocks that differ not at all or just slightly like this one.

Reply:
Good idea.